desktop/core/src/desktop/lib/importer/api.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from functools import wraps
+
+from rest_framework import status
+from rest_framework.decorators import api_view, parser_classes
+from rest_framework.parsers import JSONParser, MultiPartParser
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from desktop.lib.importer import operations
+from desktop.lib.importer.serializers import GuessFileMetadataSerializer, LocalFileUploadSerializer, PreviewFileSerializer
+
+LOG = logging.getLogger()
+
+
+# TODO: Improve error response further with better context -- Error UX Phase 2
+def api_error_handler(view_fn):
```

Comment:
are we adding an error handler specific to `importer`? Does it make sense to have one at the global level?

Reply:
Yes @amitsrivastava , the idea for global error handling makes sense and is still in scope from a long time which needs to be done soon. I'll work on it in some free time.

For now, I was sticking with error handling decorators and added a TODO for Phase 2 for global handling.

Phase 1 was ad-hoc error messages fixing.